Add several architectures and their debian images

radimkohout opened the pull request:
This pull request adds buster and bullseye debian for riscv64, mips64le, i386 and amd64(I know, that i386 and amd64 you can use without this, but with this action, it is much easier to build several platforms.

uraimo commented:
Hi, thanks, can you also add at least one new arch pair [here](https://github.com/uraimo/run-on-arch-action/blob/master/.github/workflows/test.yml#L30) for the new architectures to be sure that the emulation works? Thx.

uraimo commented:
Unstable likely fails because git was not installed [here](https://github.com/uraimo/run-on-arch-action/blob/master/.github/workflows/test.yml#L83), adding unstable to the options should fix it.
No idea about the other two, amd64 should not even use qemu-static to create the container, while that x86_64 is the name of static driver qemu has.

radimkohout commented:
I'll try to address this issues in next few days, stay tuned for update :)

radimkohout commented:
Ok, all issues fixed, things done to solve the issues:
1) Added `ENTRYPOINT ["linux32", "--"]` to i386 Dockerfiles (see  https://github.com/ioft/dockerhub/issues/1 ), also added note to the README "Contributing" part
2) Added override to amd64(actually it's x86_64) and i386(actually i686).
3) Added unstable override to "Assert container uses distro" for riscv64.

radimkohout commented:
Update: Added alpine_latest for i386 from https://github.com/uraimo/run-on-arch-action/pull/59

uraimo commented:
After a few other PRs are merged, we'll merge this too, sorry for the delay, thanks again.

radimkohout commented:
Ok, thanks. I've already made a version for my private use, so don't worry.  But I'll migrate here as soon as there will be a new version.

Flole998 commented on .github/workflows/test.yml:
```diff
@@ -80,7 +90,7 @@ jobs:
         install: |
           case "${{ matrix.distro }}" in
-            ubuntu*|jessie|stretch|buster|bullseye)
+            ubuntu*|jessie|stretch|buster|bullseye|unstable|experimantal)
```
Is the typo (experimantal instead of experimental) intentional here?

radimkohout replied:
Thanks, fixed.

martin-g commented:
I also wanted to create an issue asking for adding `x86_64/amd64` to the available architectures. It would be much easier to use the GHA matrix feature to test on both "native" arch (i.e. amd64) and non-native (emulated) ones. I see other people already had this idea!
Looking forward this PR to get some love soon and be merged!

lalten commented:
@uraimo what's the plan here? Anything still blocking this?

uraimo commented:
Just an update on this, support for riscv64 has been added and a base_image parameter for completely custom images (any arch any distro) has been included in 2.5.0 that I've just tagged.

Since I was also planning to refactor the Dockerfile mechanism (too many files) in the next major release, that I would ideally complete next week, this PR will be reworked to become part of the new default images configuration, thanks to @radimkohout and all those that participated.
